Services/Common/OJS.Services.Common.Data/Implementations/SubmissionsForProcessingCommonDataService.cs

```diff
@@ -22,16 +23,43 @@ public SubmissionsForProcessingDataService(DbContext submissionsForProcessing)
             .Where(s => s.SubmissionId == submissionId)
             .FirstOrDefaultAsync();
 
-    public IQueryable<SubmissionForProcessing> GetAllUnprocessed() =>
+    public IQueryable<SubmissionForProcessing> GetAllUnprocessedAndNotProcessing() =>
         this.DbSet
             .Where(sfp => !sfp.Processed && !sfp.Processing);
 
+    public IQueryable<SubmissionForProcessing> GetAllUnprocessed() =>
+        this.DbSet
+            .Where(sfp => !sfp.Processed);
+
+    public async Task<int> GetAllUnprocessedCount()
+        => await this
+            .GetAllUnprocessed()
+            .CountAsync();
+
     public async Task<IEnumerable<int>> GetIdsOfAllProcessing() =>
         await this.DbSet
             .Where(sfp => sfp.Processing && !sfp.Processed)
             .Select(sfp => sfp.Id)
             .ToListAsync();
 
+    public async Task<IEnumerable<TServiceModel>> GetAllProcessing<TServiceModel>()
+        => await this.DbSet
+            .Where(sfp => !sfp.Processed && sfp.Processing)
+            .ToListAsync()
+            .MapCollection<TServiceModel>();
```

Comment:
MapCollection should be before the ToList?

Reply:
Fixed